core/src/main/kotlin/behavior/GuildScheduledEventBehavior.kt

```diff
@@ -26,17 +26,14 @@ import kotlin.contracts.contract
 public interface GuildScheduledEventBehavior : KordEntity, Strategizable {
     public val guildId: Snowflake
 
+
     /**
      * Requests all the users which are interested in this event.
      *
      * @throws RequestException if anything goes wrong during the request
      */
-    public val users: Flow<User> get() = flow {
-        kord.rest.guild.getScheduledEventUsers(guildId, id).users.forEach {
-            val userData = UserData.from(it)
-            emit(User(userData, kord, supplier))
-        }
-    }
+    @Deprecated("Use getUsersBefore or getUsersAfter")
+    public val users: Flow<User> get() = supplier.getGuildScheduledEventUsersBefore(guildId, id)
```

Comment:
Why deprecate it? There is `supplier.getGuildScheduledEventUsers()`

Reply:
because the values are hardcoded are not interchangeable

Reply:
Then you would also have to remove `EntitySupplier.getGuildScheduledEventUsers()`, right?

Reply:
I did that already (used the after function) however, it's generally discouraged to use. I refactored it a bit and removed the deprecation

Reply:
👍
Although I would probably still use the `EntitySupplier` method instead of the `Behavior` method just in case it is overriden in any Supplier implementation

Reply:
but it does use the supplier; the behavior function just calls the supplier underneath; so just to reduce code duplication

Reply:
Let me clarify:
I would use
```kotlin
supplier.getGuildScheduledEventUsers(guildId, id, withMember = true)
```
and not one of these
```kotlin
supplier.getGuildScheduledEventUsersAfter(guildId, id, withMember = true)
supplier.getGuildScheduledEventUsersBefore(guildId, id, withMember = true)
this.getGuildScheduledEventUsersAfter()
this.getGuildScheduledEventUsersBefore()
````
Otherwise `EntitySupplier.getGuildScheduledEventUsers()` wouldn't be necessary since it isn't used elsewhere.


#

Unrelated but also noticed that it's `withMember` in some places and `withMembers` in other places.

Reply:
Depends on the endpoint's naming schema.

Supplier isn't private in anyway; so if we don't use it elsewhere doesn't mean we restrict it's use

The functions in the instance are just convincance methods it's unlikely you are going to do 

```kt
this.this.getGuildScheduledEventUsersAfter()
```
and specify the id and guild each time

Reply:
I think we are talking past each other, I'll try to be clear now:

I would suggest having a call chain like this (arrows representing which methods/getters call others):
```
GuildScheduledEventBehavior                EntitySupplier

users -----------------------------------> getGuildScheduledEventUsers() ----------+
                                                                                   |
getGuildScheduledEventUsersAfter()  ---->  getGuildScheduledEventUsersAfter()  <---+

getGuildScheduledEventUsersBefore()  --->  getGuildScheduledEventUsersBefore()
```
instead of this:
```
GuildScheduledEventBehavior                EntitySupplier

users ------+                              getGuildScheduledEventUsers() ----------+
            |                                                                      |
            v                                                                      |
getGuildScheduledEventUsersAfter()  ---->  getGuildScheduledEventUsersAfter()  <---+

getGuildScheduledEventUsersBefore()  --->  getGuildScheduledEventUsersBefore()
```
This way `GuildScheduledEventBehavior` would just delegate to the Suppliers and let them handle the fallback in case of all users (the fallback 'logic' is not duplicated), feels like a better separation of concerns to me.

Reply:
> Depends on the endpoint's naming schema.

If you mean `withMember` vs `withMembers`: it's literally `withMember` in `EntitySupplier.getGuildScheduledEventUsers[Before|After]()` and `withMembers` in `GuildScheduledEventBehavior.getGuildScheduledEventUsers[Before|After]()`

Reply:
> I would suggest having a call chain like this (arrows representing which methods/getters call others):

You are looking @ outdated  file; users actually the after function of the guild itself so user calls the function defined in the same file.
I Fixed the typo in regards to the parameter name